x/warp/contract_warp_handler.go

```diff
@@ -93,22 +93,22 @@ func handleWarpMessage(accessibleState contract.AccessibleState, input []byte, s
 	return res, remainingGas, nil
 }
 
-type addressedPayloadHandler struct{}
+type addressedCallHandler struct{}
 
-func (addressedPayloadHandler) packFailed() []byte {
+func (addressedCallHandler) packFailed() []byte {
 	return getVerifiedWarpMessageInvalidOutput
 }
 
-func (addressedPayloadHandler) handleMessage(warpMessage *warp.Message) ([]byte, error) {
-	addressedPayload, err := warpPayload.ParseAddressedPayload(warpMessage.UnsignedMessage.Payload)
+func (addressedCallHandler) handleMessage(warpMessage *warp.Message) ([]byte, error) {
+	addressedCall, err := avalancheWarpPayload.ParseAddressedCall(warpMessage.UnsignedMessage.Payload)
 	if err != nil {
-		return nil, fmt.Errorf("%w: %s", errInvalidAddressedPayload, err)
+		return nil, fmt.Errorf("%w: %s", errInvalidAddressedCall, err)
 	}
 	return PackGetVerifiedWarpMessageOutput(GetVerifiedWarpMessageOutput{
 		Message: WarpMessage{
 			SourceChainID:       common.Hash(warpMessage.SourceChainID),
-			OriginSenderAddress: addressedPayload.SourceAddress,
-			Payload:             addressedPayload.Payload,
+			OriginSenderAddress: common.BytesToAddress(addressedCall.SourceAddress),
```

Comment:
Will make a follow up PR to switch from converting the bytes to an address to exposing the byte array directly to the caller and changing the Warp Solidity contract interface.